core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockReader.java

```diff
@@ -315,13 +329,19 @@ private void updateBlockWriter(long offset) throws IOException {
     try {
       if (mBlockWriter == null && offset == 0 && !mBlockMeta.isNoCache()) {
         BlockStoreLocation loc = BlockStoreLocation.anyDirInTier(mStorageTierAssoc.getAlias(0));
-        String blockPath = mLocalBlockStore
-            .createBlock(mBlockMeta.getSessionId(), mBlockMeta.getBlockId(), loc,
-                mInitialBlockSize).getPath();
-        mBlockWriter = new LocalFileBlockWriter(blockPath);
+        mLocalBlockStore.createBlock(mBlockMeta.getSessionId(), mBlockMeta.getBlockId(), loc,
+            mInitialBlockSize);
+        mBlockWriter = mLocalBlockStore.getBlockWriter(
+            mBlockMeta.getSessionId(), mBlockMeta.getBlockId());
       }
-    } catch (IOException | BlockAlreadyExistsException | WorkerOutOfSpaceException e) {
+    } catch (BlockAlreadyExistsException e) {
       // This can happen when there are concurrent UFS readers who are all trying to cache to block.
+      LOG.warn(
```

Comment:
are the warn and debug flipped? The comment suggests that `BlockAlreadyExistsException` is normal and expected where there are concurrent UFS readers, so we want to avoid spamming the logs with `warn` messages in that case